activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -341,7 +341,7 @@ def raise_record_not_found_exception!(ids = nil, result_size = nil, expected_siz
         raise RecordNotFound.new(error, name, key, ids)
       else
         error = +"Couldn't find all #{name.pluralize} with '#{key}': "
-        error << "(#{ids.join(", ")})#{conditions} (found #{result_size} results, but was looking for #{expected_size})."
+        error << "(#{ids.respond_to?(:join) ? ids.join(", ") : ids})#{conditions} (found #{result_size} results, but was looking for #{expected_size})."
```

Comment:
How about just `Array(ids)` instead?

```suggestion
        error << "(#{Array(ids).join(", ")})#{conditions} (found #{result_size} results, but was looking for #{expected_size})."
```

Reply:
Okey it is caused by finder_methods.rb#L339, `Array({ foo: "bar"} ).size # => 2`.

https://github.com/rails/rails/blob/ccf85559bd43e19a95e04b4b2873bf21d01a1416/activerecord/lib/active_record/relation/finder_methods.rb#L339-L342

It should be `elsif Array.wrap(ids).size == 1 `.

Reply:
I change `Array(ids).size == 1` as your mentioned, and revert  `ids.respond_to?(:join) ? ids.join(", ") : ids` to `ids.join(", ")`